client/cli/src/runner.rs

```diff
@@ -116,11 +116,18 @@ impl<C: SubstrateCli> Runner<C> {
 	/// Create a new runtime with the command provided in argument
 	pub fn new<T: CliConfiguration>(cli: &C, command: &T) -> Result<Runner<C>> {
 		let tokio_runtime = build_runtime()?;
+		let runtime_handle = tokio_runtime.handle().clone();
 
 		let task_executor = {
-			let runtime_handle = tokio_runtime.handle().clone();
-			Arc::new(move |fut| {
-				runtime_handle.spawn(fut);
+			Arc::new(move |fut, task_type| {
+				match task_type {
+					TaskType::Async => { runtime_handle.spawn(fut); }
+					TaskType::Blocking => {
+						runtime_handle.spawn( async move {
+							tokio::task::spawn_blocking(move || futures::executor::block_on(fut))
```

Comment:
Let's add some comment (feel free to put something else)

```suggestion
							// `spawn_blocking` is looking for the current runtime, and as such has to be called
							// from within `spawn`.
							tokio::task::spawn_blocking(move || futures::executor::block_on(fut))
```